packages/amplify-category-storage/src/provider-utils/awscloudformation/cdk-stack-builder/s3-stack-transform.ts

```diff
@@ -12,28 +13,31 @@ import {
 } from 'amplify-cli-core';
 import { formatter, printer } from 'amplify-prompts';
 import * as fs from 'fs-extra';
+import os from 'os';
 import * as path from 'path';
 import * as vm from 'vm2';
-import os from 'os';
 import { S3PermissionType, S3UserInputs } from '../service-walkthrough-types/s3-user-input-types';
-import { canResourceBeTransformed, S3CFNDependsOn, S3CFNPermissionType, S3InputState } from '../service-walkthroughs/s3-user-input-state';
+// eslint-disable-next-line import/no-cycle
+import {
+  canResourceBeTransformed, S3CFNDependsOn, S3CFNPermissionType, S3InputState,
+} from '../service-walkthroughs/s3-user-input-state';
+// eslint-disable-next-line import/no-cycle
```

Comment:
we should log a work item to refactor those cycles. e.g. by extracting those functions to 3rd file that's then used by both modules.

Reply:
yeah , I was thinking of discussing with you for these dependency files. these are present everywhere in our code